src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -946,13 +972,53 @@ Real64 GLHEVert::doubleIntegral(std::shared_ptr<GLHEVertSingle> const &bh_i, std
 
 //******************************************************************************
 
+void GLHEVert::calcLongTimestepGFunctions(EnergyPlusData &state)
+{
+    switch (this->gFuncCalcMethod) {
+    case GFuncCalcMethod::UniformHeatFlux:
+        this->calcUniformHeatFluxGFunctions(state);
+        break;
+    case GFuncCalcMethod::UniformBoreholeWallTemp:
+        this->calcUniformBHWallTempGFunctions();
+        break;
+    default:
+        assert(false);
+    }
+}
+
+//******************************************************************************
+
+void GLHEVert::calcUniformBHWallTempGFunctions()
+{
+    // construct boreholes vector
+    std::vector<gt::boreholes::Borehole> boreholes;
+    for (auto &bh : this->myRespFactors->myBorholes) {
+        boreholes.emplace_back(bh->props->bhLength, bh->props->bhTopDepth, bh->props->bhDiameter / 2.0, bh->xLoc, bh->yLoc);
+    }
+
+    // convert time to a std::vector from an Array1D
+    std::vector<double> time;
+    for (auto &v : this->myRespFactors->time) {
+        time.push_back(v);
+    }
+
+    // Obtain number of segments by adaptive discretization
+    gt::segments::adaptive adptDisc;
+    int nSegments = adptDisc.discretize(this->bhLength, this->totalTubeLength);
+
+    this->myRespFactors->GFNC = gt::gfunction::uniform_borehole_wall_temperature(boreholes, time, this->soil.diffusivity, nSegments);
+}
```

Comment:
The adaptive discretization struct is called to choose a number of segments for UBHWTcalc that will approximate a UIFT g-function.